extensions/_modules/boto_ec2.py

```diff
@@ -1815,7 +1815,7 @@ def detach_volume(volume_id=None, instance_id=None, device=None, force=False,
             return False
     elif tags:
         try:
-            volumes = conn.get_all_volumes(Filters = [{'Name': tag_name, 'Values':[tag_value]} for tag_name, tag_value in tags.items()])
+            volumes = conn.get_all_volumes(filters = {'tag:{}'.format(tag_name):'{}'.format(tag_value)})
```

Comment:
Salt still supports Python 2.6 so you need to include the `0` index in the format string.